browser/tor/tor_profile_manager_browsertest.cc

```diff
@@ -99,8 +102,32 @@ class TorProfileManagerTest : public InProcessBrowserTest {
   MockTorLauncherFactory* GetTorLauncherFactory() {
     return &MockTorLauncherFactory::GetInstance();
   }
+
+  void Relaunch(const base::CommandLine& new_command_line) {
+    base::LaunchProcess(new_command_line, base::LaunchOptionsForTest());
+  }
 };
 
+IN_PROC_BROWSER_TEST_F(TorProfileManagerTest, LaunchWithTorUrl) {
+  // We should start with one normal window.
+  ASSERT_EQ(1u, chrome::GetTabbedBrowserCount(browser()->profile()));
+
+  // Run with --tor switch and a URL specified.
+  base::FilePath test_file_path = ui_test_utils::GetTestFilePath(
+      base::FilePath(), base::FilePath().AppendASCII("empty.html"));
+  base::CommandLine new_command_line(GetCommandLineForRelaunch());
+  new_command_line.AppendSwitch(switches::kTor);
+  new_command_line.AppendArgPath(test_file_path);
+
+  Relaunch(new_command_line);
+
+  // There should be one normal and one Tor window now.
```

Comment:
`SecondLaunchWithIncognitoUrl` is disabled on Windows and Linux upstream though
```
// ChromeMainTest.SecondLaunchWithIncognitoUrl is flaky on Win and Linux.
// http://crbug.com/130395
#if defined(OS_WIN) || defined(OS_LINUX) || defined(OS_CHROMEOS)
#define MAYBE_SecondLaunchWithIncognitoUrl DISABLED_SecondLaunchWithIncognitoUrl
#else
#define MAYBE_SecondLaunchWithIncognitoUrl SecondLaunchWithIncognitoUrl
#endif
```

Reply:
For what it's worth, this browser test worked just fine for me on Linux but I disabled it in 2a0cf410345001bc681a9c34fdc99ca13a9e70b7.